EngineBay.Auditing.Tests/AuditInterceptorTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿namespace EngineBay.Auditing.Tests
+{
+    using EngineBay.Auditing.Tests.FakeAuditableModel;
+    using EngineBay.Persistence;
+    using Newtonsoft.Json;
+    using Xunit;
+
+    public class AuditInterceptorTests : BaseTestWithFullAuditedDb<FakeDbContext>
+    {
+        private readonly List<FakeModel> fakeModels;
+        private readonly List<ApplicationUser> applicationUsers;
+
+        public AuditInterceptorTests()
+            : base(nameof(AuditInterceptorTests))
+        {
+            var applicationUsersPath = Path.GetFullPath(@"./TestData/application-users.json");
+
+            var tempApplicationUsers = JsonConvert.DeserializeObject<List<ApplicationUser>>(File.ReadAllText(applicationUsersPath));
+            if (tempApplicationUsers is null)
+            {
+                throw new ArgumentException("Application users not found at " + applicationUsersPath);
+            }
+
+            this.applicationUsers = tempApplicationUsers;
+
+            var fakeModelPath = Path.GetFullPath(@"./TestData/fake-models.json");
+            var tempFakeModels = JsonConvert.DeserializeObject<List<FakeModel>>(File.ReadAllText(fakeModelPath));
+            if (tempFakeModels is null)
+            {
+                throw new ArgumentException("Fake models not found at " + fakeModelPath);
+            }
+
+            this.fakeModels = tempFakeModels;
+
+            this.DbContext.RemoveRange(this.DbContext.ApplicationUsers);
+            this.DbContext.AddRange(this.applicationUsers);
+            this.DbContext.SaveChanges();
+            this.DbContext.RemoveRange(this.DbContext.FakeModels);
```

Comment:
Add a tolist here. I don't like the fact that you are modifying the ApplicationUsers whiile querying it